### ISSUE TITLE
Histology file fields

### ISSUE THREAD

**Issue by komalsrathi** [d3b-center/d3b-bixu-data-assembly#16]
Comparing the primary sample information for Patient 43 and 35 from v11 to the corresponding progression samples in the new histology file [histologies_v11-base-add-BS_AG4BP2PM.tsv](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-data-assembly/files/62ac80c86c240b371798b30a/), I think some of the fields have missing information. Is this because there is no information available?

1. `pathology_diagnosis` should be `High-grade glioma/astrocytoma (WHO grade III/IV)` (?). This is only for the +1 patient.
2. `broad_histology` should be `Diffuse astrocytic and oligodendroglial tumor` (?). This is only for the +1 patient.
3. `short_histology` should be `HGAT` (?). This is only for the +1 patient.
4. ~`RNA_library` seems to have moved to `PFS_days` for `BS_A7Y1Y314` (`C3506976`). This is from `histologies_v11-base-add-BS_AG4BP2PM.tsv`~ Seems to have been resolved. 
5. `molecular_subtype` seems to be missing entirely from [histologies_v11-base-add-BS_AG4BP2PM.tsv](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-data-assembly/files/62b0d11e6c240b3717abcb58/) and [histologies_v11-base-add-BS_HSXARQ1K.tsv](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-data-assembly/files/62ab6ccf6894ba72bd33573a/). I think it was present in v11 histologies.tsv (but not in histologies-base.tsv). So they are NA across all samples.
6. Just noticed the following discrepancy, two samples `BS_J4E9SW51` and `BS_H1XPVS9A` (both correspond to `C334437`) are annotated as `LGAT` in v10/v11 but `HGAT` in data assembly histology files:

```
# OpenPedCan v10 histology
v10_histology <- read_tsv('data/OpenPedCan-analysis/data/v10/histologies.tsv')
v10_histology %>%
  filter(Kids_First_Biospecimen_ID %in% c("BS_J4E9SW51", "BS_H1XPVS9A")) %>%
  pull(short_histology)
[1] "Low-grade astrocytic tumor" "Low-grade astrocytic tumor"

# OpenPedCan v11 histology
v11_histology <- read_tsv('data/OpenPedCan-analysis/data/v11/histologies.tsv')
v11_histology %>%
  filter(Kids_First_Biospecimen_ID %in% c("BS_J4E9SW51", "BS_H1XPVS9A")) %>%
  pull(short_histology)
[1] "LGAT" "LGAT"

# data assembly file generated with BS_HSXARQ1K
dat <- read_tsv('histologies_v11-base-add-BS_HSXARQ1K.tsv')
dat %>%
  filter(Kids_First_Biospecimen_ID %in% c("BS_J4E9SW51", "BS_H1XPVS9A")) %>%
  pull(short_histology)
[1] "HGAT" "HGAT"

# data assembly file with BS_AG4BP2PM
dat <- read_tsv('histologies_v11-base-add-BS_AG4BP2PM.tsv')
dat %>%
  filter(Kids_First_Biospecimen_ID %in% c("BS_J4E9SW51", "BS_H1XPVS9A")) %>%
  pull(short_histology)
[1] "HGAT" "HGAT"
```

I manually filled these for the report generation so don't need these. Just wanted to make a note of it.

cc @aadamk

**Comment by komalsrathi** [comment]
Found in the data assembly file which I have linked in the ticket - the same file is being used for reporting code.

**Comment by zhangb1** [comment]
Yeah, I don't have those informations when doing the data assembly so I didn't add those.

**Comment by aadamk** [comment]
For 1, this may be reliant on migrating out of DT to Redcap for future PNOC trials so that pathology free text gets mapped to the CBTN term upstream by the CRU. 
For 2 and 3, we need to get the short and broad hist PR merged into the toolkit. 
For 4, do you mean that the `PFS_days` contains an `RNA_library` name for that biospecimen? 

Also @zhangb1 - for a PNOC008 recurrence sample that was completed last week, [BS_AG4BP2PM](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-data-assembly/files/62ac86666c240b3717995694/), the absolute copy number (`copy_number` field in the linked output file) is a floating point number rather than integer. It seems the `depth` field from cnvkit may have somehow been pulled into the merged output file, whereas for [BS_HSXARQ1K](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-data-assembly/files/62ab71f66894ba72bd340c63/), this is not the case - the `copy_number` field contains an integer. As such, BS_AG4BP2PM contains many events erroneously called as amplifications. Can you dig into why this may have happened and implement a fix?

**Comment by komalsrathi** [comment]
> For 4, do you mean that the `PFS_days` contains an `RNA_library` name for that biospecimen?

Yes, here is what I meant: 

```
> dat %>% filter(Kids_First_Biospecimen_ID == "BS_A7Y1Y314") %>% dplyr::select(RNA_library, PFS_days)
  RNA_library PFS_days
1        <NA> stranded
```

**Comment by komalsrathi** [comment]
But the last merged file should have both patients. Because we want to compare all patients up to that point...

**Comment by aadamk** [comment]
> We only do N+1.....

can you set things up in such a way so that you keep recursively adding on to the latest file? 
e.g. N = v11 file; N + 1 = v11 + new patient;
then, when a new patient is enrolled, the next cwl run is added on to N+1 to generate N + 2 (e.g. v11 + previously enrolled patient + new patient), and so on?

**Comment by aadamk** [comment]
and for that logic to apply to all files (snv, sv, cnv, rnaseq, fusion, etc)?

**Comment by zhangb1** [comment]
Can we setup a meeting to discuss this?  We should discuss with @yuankunzhu  about how to process that. 

We may need to wrap up  the new CWL to do that.

**Comment by zhangb1** [comment]
Also @komalsrathi  in the task here : https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-data-assembly/tasks/1ba0c658-6c10-49c8-8ec9-cf89bf5b88bf/

the float issue for BS_AG4BP2PM has been fixed.

**Comment by aadamk** [comment]
> Can we setup a meeting to discuss this? We should discuss with @yuankunzhu about how to process that.
> 
> We may need to wrap up the new CWL to do that.
ok - tomorrow's toolkit meeting should be fine.

**Comment by komalsrathi** [comment]
Thanks @zhangb1 I'll merge those files on my end.

**Comment by komalsrathi** [comment]
@jharenza I added a 6th point to this ticket - wasn't sure if this is an issue with the v10/v11 file or something during data assembly.

**Comment by jharenza** [comment]
> @jharenza I added a 6th point to this ticket - wasn't sure if this is an issue with the v10/v11 file or something during data assembly.

>Just noticed the following discrepancy, two samples BS_J4E9SW51 and BS_H1XPVS9A are annotated as LGAT in v10/v11 but HGAT in data assembly histology files:

Thanks @komalsrathi. The comparison between data assembly and v11 won't be apples to apples because the data assembly doesn't include molecular subtyping. These two samples changed following subtyping from `pathology_diagnosis == High-grade glioma/astrocytoma (WHO grade III/IV)` to `Pleomorphic xanthoastrocytoma, BRAF V600E`. LGG is the module which subtypes PXAs. However, with that being said, they are also special cases because they are "malignant pxa" according to `pathology_free_text_diagnosis`. Unfortunately, we do not have a field for grade in the histology file (aside from the LGG or HGG `pathology_diagnosis` field) and PXA can be either grade II or III. One thing we can do in the future is add a manual review of path reports + discussion with pathologists for tumors which are subtyped via LGG subtyping _and_ can be higher grades. This is something I had recently thought about, but is difficult to do perfectly with the given data, so thanks for bringing this up. These two are _likely_ high-grade PXAs, but that also begs the question of whether we want to lump them in with other HGGs. There is a new (2021) category of tumors called "Circumscribed astrocytic gliomas", separate of diffuse HGGs (of interest here), which contains: Pilocytic astrocytoma, High-grade astrocytoma with piloid features, Pleomorphic xanthoastrocytoma, Subependymal giant cell astrocytoma, Chordoid glioma, and Astroblastoma, MN1-altered and these are of mixed grade. So, I guess my advice would be to keep these two samples out of the HGG group since they are likely pretty biologically different, even though they are potentially _higher grade_ tumors. cc @aadamk
